Fixes high CPU usage after device-vibration

mayanksuman opened the pull request:
This commit introduces separate read and write QSocketNotifier for
subdevices. Write notifier is required for sending data to HID
subdevice. Read notifier for HID subdevice will help in reading data
from device like battery percentage and configuring the spotlight device
like configuring it to send long press events for Next and previous
button (See #71).

This commit also fixes #133 as read notifier for HID subdevice do not
send activated signal continuosly after a write operation. Rather the
on activated signal, the device is read (However it currently does not
        do anything with the read value).

Fixes #133

jahnf commented:
Thank you. Will do a review in the upcoming week 👍

jahnf commented on src/device.cc:
```diff
@@ -238,14 +264,53 @@ std::shared_ptr<SubHidrawConnection> SubHidrawConnection::create(const DeviceSca
+    enableWrite();
+    const auto notifier = socketWriteNotifier();
+    res = ::write(notifier->socket(), hidppMsg.data(), hidppMsg.length());
+    logInfo(hid) << "Write " << hidppMsg.toHex();
```
Should be `logDebug` instead of `logInfo` also more detailed: e.g.:
```suggestion
    logDebug(hid) << "Write " << hidppMsg.toHex() << "from" << connection.path(); 
```

mayanksuman replied:
Hi! the change has been implemented.

jahnf commented on src/spotlight.cc:
```diff
@@ -288,22 +289,61 @@ void Spotlight::onEventDataAvailable(int fd, SubEventConnection& connection)
+    }
+    return;
+  }
+  logInfo(hid) << "Received " << readVal.toHex();
```
Should be `logDebug` instead of `logInfo` also more detailed: e.g.:
```suggestion
  logDebug(hid) << "Received " << readVal.toHex() << "from" << connection.path(); 
```

mayanksuman replied:
The change has been implemented.

jahnf commented on src/device.h:
```diff
@@ -127,6 +127,11 @@ class SubDeviceConnection : public QObject
+  void disableWrite(); // disable sending data
+  void enableWrite(); // enable sending data
+
+  ssize_t sendData(QByteArray hidppMsg);                          // Send HID++ Message to HIDraw connection
```
```suggestion
  ssize_t sendData(const QByteArray& hidppMsg);                          // Send HID++ Message to HIDraw connection
```

jahnf commented on src/device.cc:
```diff
@@ -238,14 +264,53 @@ std::shared_ptr<SubHidrawConnection> SubHidrawConnection::create(const DeviceSca
 }
+
+// -------------------------------------------------------------------------------------------------
+ssize_t SubDeviceConnection::sendData(QByteArray hidppMsg)
```
```suggestion
ssize_t SubDeviceConnection::sendData(const QByteArray& hidppMsg)
```

mayanksuman replied:
The change has been implemented.

jahnf commented on src/device.cc:
```diff
@@ -238,14 +264,53 @@ std::shared_ptr<SubHidrawConnection> SubHidrawConnection::create(const DeviceSca
+
+
+// -------------------------------------------------------------------------------------------------
+ssize_t SubDeviceConnection::sendData(const uint8_t hidppMsg[], size_t hidppMsgLen)
```
```suggestion
ssize_t SubDeviceConnection::sendData(const void* hidppMsg, size_t hidppMsgLen)
```

mayanksuman replied:
The change has been implemented.

jahnf commented on src/device.cc:
```diff
@@ -238,14 +264,53 @@ std::shared_ptr<SubHidrawConnection> SubHidrawConnection::create(const DeviceSca
+// -------------------------------------------------------------------------------------------------
+ssize_t SubDeviceConnection::sendData(const uint8_t hidppMsg[], size_t hidppMsgLen)
+{
+  QByteArray hidppMsgArr;
+  for(size_t i=0; i<hidppMsgLen; i++)
+    hidppMsgArr.append(hidppMsg[i]);
```
```suggestion
  const QByteArray hidppMsgArr(reinterpret_cast<const char*>(hidppMsg), hidppMsgLen);
```

mayanksuman replied:
The change has been implemented in the latest commit.

jahnf reviewed (changes requested):
Hi @mayanksuman, sorry for the delay. I finally came to review the PR - I only have some smaller suggestions.

mayanksuman commented:
Hi @jahnf, Thanks for the review. I have implemented all the changes. However during rebasing, I rewrote the history by mistake. 

I have made a new pull request (#137 ) with clean history (and all the changes recommended).

I am closing this pull request to avoid any confusion. Please review the new PR #137.
